examples/eady_turbulence.jl

```diff
@@ -5,306 +5,398 @@
+vᵢ(x, y, z) = Ũ * Ξ(z)
+bᵢ(x, y, z) = B̃ * Ξ(z)
 
-# # Diagnostics and output
-#
-# Diagnostics that return the maximum absolute value of `u, v, w` by calling
-# `umax(model), vmax(model), wmax(model)`:
+set!(model, u=uᵢ, v=vᵢ, b=bᵢ)
+
+# We subtract off any residual mean velocity to avoid exciting domain-scale
+# inertial oscillations. We use a `sum` over the entire `parent` arrays or data
+# to ensure this operation is efficient on the GPU (set `architecture = GPU()`
+# in `IncompressibleModel` constructor to run this problem on the GPU if one
+# is available).
 
-umax = FieldMaximum(abs, model.velocities.u)
-vmax = FieldMaximum(abs, model.velocities.v)
-wmax = FieldMaximum(abs, model.velocities.w)
+Ū = sum(model.velocities.u.data.parent) / (grid.Nx * grid.Ny * grid.Nz)
+V̄ = sum(model.velocities.v.data.parent) / (grid.Nx * grid.Ny * grid.Nz)
 
-# Set up output. Here we output the velocity and buoyancy fields at intervals of one day.
-output_writer = JLD2OutputWriter(model, merge(model.velocities, model.tracers);
-                                 time_interval=day, prefix=output_filename_prefix,
-                                 force=true, max_filesize=10GiB)
+model.velocities.u.data.parent .-= Ū
+model.velocities.v.data.parent .-= V̄
 
-# # The `TimeStepWizard`
+# # Simulation set-up
+#
+# We set up a simulation and run it for 10 days.
+# We then stop the simulation and add a JLD2OutputWriter that saves the vertical
+# velocity, vertical vorticity, and divergence every 2 iterations.
+# We then continue the run for 100 iterations and plot the resulting 50 frames of
+# saved data in an animation.
+#
+# ## The `TimeStepWizard`
 #
 # The TimeStepWizard manages the time-step adaptively, keeping the CFL close to a
 # desired value.
 
-wizard = TimeStepWizard(cfl=0.05, Δt=20.0, max_change=1.1, max_Δt=min(1/10f, σᵇ/10))
+using Oceananigans.Utils: minute
 
-# # Vertical vorticity and divergence for plotting purposes
-#
-# We also create objects for computing the vertical vorticity and divergence
-# for plotting purposes.
-
-u, v, w = model.velocities
-ζ = Field(Face, Face, Cell, model.architecture, model.grid, TracerBoundaryConditions(grid))
-δ = Field(Cell, Cell, Cell, model.architecture, model.grid, TracerBoundaryConditions(grid))
+## Calculate absolute limit on time-step using diffusivities and 
+## background velocity.
+Ū = background_parameters.α * grid.Lz
 
-vertical_vorticity = Computation(∂x(v) - ∂y(u), ζ)
-        divergence = Computation(-∂z(w), δ)
+max_Δt = min(grid.Δx / Ū, grid.Δx^4 / κ₄h, grid.Δz^2 / κ₂z)
 
-ζmax = FieldMaximum(abs, ζ)
-δmax = FieldMaximum(abs, δ)
+wizard = TimeStepWizard(cfl=1.0, Δt=0.1*max_Δt, max_change=1.1, max_Δt=max_Δt)
 
-# # Plot setup
+# ## A progress messenger
 #
-# Set `makeplot = true` to make plots as the model runs.
+# We write a function that prints out a helpful progress message while the simulation runs.
 
-makeplot = false
+using Printf
+using Oceananigans.Diagnostics: AdvectiveCFL
```

Comment:
I feel like import statements should go at the very top of the file but maybe this is more instructive/useful in an example.

Reply:
I agree with that philosophy for source code, but for examples I think it's pedagogical to put the import statements next to the place where the names are used. I think this makes the example more readable, eg they can be understood by reading top to bottom without having to "refer" to other parts of the script. What do you think about that?